kernel/src/expressions/transforms.rs

```diff
@@ -118,6 +118,15 @@ pub trait ExpressionTransform<'a> {
         self.recurse_into_pred_binary(pred)
     }
 
+    /// Called for each [`VariadicExpression`] encountered during the traversal. Implementations can
+    /// call [`Self::recurse_into_expr_variadic`] if they wish to recursively transform the children.
+    fn transform_expr_variadic(
```

Comment:
are the expression transform changes sufficiently tested?

Reply:
I added tests for `transform_expr_variadic`.